apps/src/code-studio/components/AssetManager.jsx

```diff
@@ -106,6 +109,14 @@ export default class AssetManager extends React.Component {
   onUploadError = (status) => {
     this.setState({statusMessage: 'Error uploading file: ' +
       getErrorMessage(status)});
+    firehoseClient.putRecord(
+      {
+        study: 'project-data-integrity',
+        study_group: 'v3',
+        event: 'asset-upload-error',
+        project_id: this.props.projectId
```

Comment:
I'd recommend also logging the `status` (assuming that is the numeric http status code) as `data_int`. You never know exactly what info you'll need while debugging, but it seems like it could be a good hint as to why the image upload failed.